src/core/pusher.ts

```diff
@@ -116,26 +132,46 @@ export default class Pusher {
           channel.handleEvent(event);
         }
       }
+
       // Emit globally [deprecated]
       if (!internal) {
         this.global_emitter.emit(event.event, event.data);
       }
     });
+
     this.connection.bind('connecting', () => {
       this.channels.disconnect();
     });
+
     this.connection.bind('disconnected', () => {
       this.channels.disconnect();
     });
+
     this.connection.bind('error', err => {
       Logger.warn(err);
     });
+  }
 
-    Pusher.instances.push(this);
-    this.timeline.info({ instances: Pusher.instances.length });
+  /**
+   * Allows you to switch Pusher cluster without
+   * losing all the channels/subscription binding
+   * as this is internally managed by the SDK.
+   */
+  switchCluster(options: ClusterOptions) {
+    // Automatically disconnect the current connection
+    this.disconnect();
```

Comment:
I was wondering if "should" persist the previous WebSocket connection until the new connection is established, or it matters less? 
Supporting this use-case might add a bit of complexity to the code to support two instances of  `ConnectionManager`.
I'd love to have your feedback.